From c44ac17a04bdf628a8a0371241ecaff86879b12b Mon Sep 17 00:00:00 2001
From: Sohail Ahmad <sohail@indionetworks.com>
Date: Mon, 5 Dec 2022 02:58:36 +0000
Subject: [PATCH 29/55] dnsmasq: enabled EDNS configuration options- CPE ID and
 subnet

Enables EDNS CPE ID and client subnet options in dnsmasq configuration, supporting DNS-based client identification and geolocation features.

Signed-off-by: Sohail Ahmad <sohail@indionetworks.com>
---
 package/network/services/dnsmasq/files/dnsmasq.init | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/package/network/services/dnsmasq/files/dnsmasq.init b/package/network/services/dnsmasq/files/dnsmasq.init
index 8acc40d963..cdcf856439 100755
--- a/package/network/services/dnsmasq/files/dnsmasq.init
+++ b/package/network/services/dnsmasq/files/dnsmasq.init
@@ -1066,6 +1066,10 @@ dnsmasq_start()
 	append_parm "$cfg" "max_cache_ttl" "--max-cache-ttl"
 	append_parm "$cfg" "pxe_prompt" "--pxe-prompt"
 	append_parm "$cfg" "tftp_unique_root" "--tftp-unique-root"
+
+	append_parm "$cfg" "addcpe" "--add-cpe-id"
+	append_parm "$cfg" "addsubnet" "--add-subnet"
+
 	config_list_foreach "$cfg" "pxe_service" append_pxe_service
 	config_get DOMAIN "$cfg" domain
 
-- 
2.34.1

